scripts/nasa/cmip6_sea_level/process.py

```diff
@@ -0,0 +1,312 @@
+        ],
+        _SVID: ('SSP585', 2),
+    },
+    'low': {
+        _MMETHOD: 'IPCC_LowConfidence',
+    },
+    'medium': {
+        _MMETHOD: 'IPCC_MediumConfidence',
+    },
+}
+
+
+def _fname(file_path):
+    return os.path.basename(file_path)[:-3]  # strip '.nc'
+
+
+def parse_sv_info(file_path):
+    fname = _fname(file_path)
+    sv_info = {
+        _PV: [],
+    }
+    sv_id_parts = {}
+    for part in fname.split('_'):
+        if part not in _SV_CONFIG:
+            continue
+        part_info = _SV_CONFIG[part]
+
+        if _PV in part_info:
+            sv_info[_PV].extend(part_info[_PV])
+
+        if _SVID in part_info:
+            id_pair = part_info[_SVID]
+            assert id_pair[1] not in sv_id_parts, sv_id_parts
+            sv_id_parts[id_pair[1]] = id_pair[0]
+
+        if _NCVAR in part_info:
+            assert _NCVAR not in sv_info
+            sv_info[_NCVAR] = part_info[_NCVAR]
+
+        if _UNIT in part_info:
+            assert _UNIT not in sv_info
+            sv_info[_UNIT] = part_info[_UNIT]
+
+        if _MMETHOD in part_info:
+            assert _MMETHOD not in sv_info
+            sv_info[_MMETHOD] = part_info[_MMETHOD]
+
+    assert _NCVAR in sv_info, 'Could not map to an ncvar for: ' + fname
+    sv_info[_SVID] = '_'.join([sv_id_parts[i] for i in sorted(sv_id_parts)])
+    return sv_info
+
+
+###
+### Code specific to SeaLevel NC files
+###
+
+
+def to_place_dcid(location, prefix=''):
+    if location == -1:
+        return prefix + 'Earth'
+    elif location < 1000000000:
+        return prefix + 'psmslId/' + str(location)
+    else:
+        # Global grid location IDs are formatted “10MMM0NNN0” where MMM and NNN
+        # are the whole degrees latitude and longitude.  MMM goes from 0-180
+        # while NNN goes from 0-360
+        lstr = str(location)
+        lat = 90 - int(lstr[2:5])
+        lon = int(lstr[6:9])
+        if lon > 180:
+            lon -= 360
+        return prefix + 'grid_1/' + str(lat) + '_' + str(lon)
+
+
+def to_contained_places(location, id2cip):
+    if location == -1:
+        return ''
+    else:
+        dcid = to_place_dcid(location)
+        places = ['dcid:Earth']
+        if dcid in id2cip:
+            places.extend(['dcid:' + x for x in id2cip[dcid]])
+        return ','.join(places)
+
+
+def to_place_type(location):
+    if location == -1:
+        return 'dcs:Place'
+    elif location < 1000000000:
+        return 'dcs:TideGaugeStation'
+    else:
+        return 'dcs:GeoGridPlace_1Deg'
+
+
+def to_sv(quantile, sv_info):
+    if quantile == 0.0:
+        prefix = 'Min'
+    elif quantile == 0.1:
+        prefix = 'Percentile10'
+    elif quantile == 0.5:
+        prefix = 'Median'
+    elif quantile == 0.9:
+        prefix = 'Percentile90'
+    elif quantile == 1.0:
+        prefix = 'Max'
+    else:
+        assert not quantile, 'Unexpected value: ' + str(quantile)
+    return 'dcid:' + prefix + '_' + sv_info[_SVID]
```

Comment:
sv_info[_SVID] returns a tuple -- maybe encapsulating this in a get would clean up specifying which tuple index to use?

Reply:
ah, i missed that this is the result of parse_sv_info and not the config. consider renaming sv_info, or adding param documentation so it's clear?

Reply:
Added param documentation.